### ISSUE TITLE
Screen Reader: Reports: The checkboxes are announced as buttons and without a state

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77466]
## Action Performed:
Prerequisites:
- the user is signed in

1. Using VoiceOver on iOS or TalkBack on Android, open the application.
2. Once the Inbox screen opens, swipe right to navigate to the Reports tab and double-tap to activate it.
3. Once the "Reports" screen opens, swipe right to navigate to the "Status" dropdown and double-tap to activate it.
4. Once the "Status" dropdown expands, swipe right to navigate to the options.
5. Note how each checkbox is announced.

## Expected Result:
Each option here behaves like a checkbox - multiple can be selected at once, so it is expected to announce them with a checkbox role and checked or not checked state.

## Actual Result:
Each option here behaves like a checkbox - multiple can be selected at once; however, they are announced without a button role and without state.

## Area issue was found in:
Reports > Filters 

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Use the appropriate accessibility properties to correctly expose the role and state, such as accessibilityRole="checkbox" and accessibilityState={{ checked: true/false }}. Ensure that these properties are updated dynamically when the selection changes. This will allow screen readers to announce the controls as checkboxes and accurately reflect their checked/unchecked state.
https://reactnative.dev/docs/accessibility#accessibilityrole
https://reactnative.dev/docs/accessibility#accessibilitystate

## Environment:
Apple iPhone 16 - iOS 18.2, Wi-Fi - Native (No mobile browser)
Vivo IQOO Z9s 5G - Android 15, Wi-Fi - Native (No mobile browser)

## Screenshots/Videos

<img width="306" height="621" alt="Image" src="https://github.com/user-attachments/assets/acd650b4-2d27-49d8-867b-83855d0f0d1c" />

<img width="304" height="616" alt="Image" src="https://github.com/user-attachments/assets/6eca252c-c737-44b8-97b1-ba53142fa580" />

https://github.com/user-attachments/assets/50e2d249-a03c-40a6-b1e5-a1bd2c44523a

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7007059
**Issue reported by:** NA
**Slack conversation:** NA
